src/module/recorder/ytdlp.rs

```diff
@@ -0,0 +1,509 @@
+                    Some(Message::ToNotify(Notification {
+                        task: task.clone(),
+                        status: TaskStatus::Done,
+                    }))
+                }
+                RecorderState::AlreadyProcessed => {
+                    info!("{} Video already processed, skipping", task_name);
+                    None
+                }
+                RecorderState::Interrupted => {
+                    info!("{} Recording failed: interrupted", task_name);
+                    Some(Message::ToNotify(Notification {
+                        task: task.clone(),
+                        status: TaskStatus::Failed,
+                    }))
+                }
+                _ => None,
+            };
+
+            if let Some(message) = message {
+                // Exit the loop if message failed to send
+                if let Err(_) = bus.send(message).await {
+                    break;
+                }
+            }
+        }
+
+        trace!("{} Status loop exited: {:?}", task_name, parser.video_status);
+
+        // Wait for threads to finish
+        let (r_wait, r_stdout, r_stderr) = futures::join!(h_wait, h_stdout, h_stderr);
+        trace!("{} Process monitor exited: {:?}", task_name, r_wait);
+        trace!("{} Stdout monitor quit: {:?}", task_name, r_stdout);
+        trace!("{} Stderr monitor quit: {:?}", task_name, r_stderr);
+
+        // Skip moving files if it didn't finish
+        if parser.video_status.state != RecorderState::Finished {
+            return Ok(());
+        }
+
+        // Move the video to the output directory
+        let frompath = parser.video_status
+            .output_file
+            .ok_or(anyhow!("yt-dlp did not emit an output file"))?;
+        let frompath = Path::new(&frompath);
+        let filename = frompath
+            .file_name()
+            .ok_or(anyhow!("Failed to get filename"))?;
+        let destpath = Path::new(&task.output_directory).join(filename);
+
+        // Try to rename the file into the output directory
+        if let Err(_) = fs::rename(frompath, &destpath) {
+            debug!(
+                "{} Failed to rename file to output, trying to copy",
+                task_name,
+            );
+
+            // Copy the file into the output directory
+            fs::copy(frompath, &destpath)
+                .with_context(|| format!("Failed to copy file to output: {:?}", destpath))?;
+            info!(
+                "{} Copied output file to {}, removing original",
+                task_name,
+                destpath.display(),
+            );
+            fs::remove_file(frompath)
+                .with_context(|| format!("Failed to remove original file: {:?}", frompath))?;
+        }
+
+        info!("{} Moved output file to {}", task_name, destpath.display());
+        Ok(())
+    }
+}
+
+pub struct YTDOutputParser {
+    video_status: VideoStatus,
+}
+
+impl YTDOutputParser {
```

Comment:
Since ytarchive and yt-dlp uses a similar method to parse the command line output, I think it'd be good if you can create an `OutputParser` trait, so both `YTDOutputParser` and `YTAOutputParser` shares the same interface.